src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/StackLayout.cs

```diff
@@ -368,9 +370,10 @@ protected internal override bool IsSignificantViewportChange(Rect oldViewport, R
 				return base.IsSignificantViewportChange(oldViewport, newViewport);
 			}
 
-			if (_uno_lastKnownRealizedElementsCount <= 1)
+			var neededElementsCountToFillNewViewport = Math.Min(_uno_lastKnownItemsCount, MajorSize(newViewport) / _uno_lastKnownAverageElementSize);
+			if (_uno_lastKnownRealizedElementsCount < neededElementsCountToFillNewViewport)
 			{
-				// Only the first item has been measured so far, this might be because the IR is within a SV and not visible yet.
+				// Only few first items have been measured so far (usually 1 or 2), this might be because the IR is within a SV and not visible yet.
```

Comment:
```suggestion
				// Only a few first items have been measured so far (usually 1 or 2), this might be because the IR is within a SV and not visible yet.
```